### PR TITLE
`<PopoverMenu.MenuItem/>`, `<PopoverMenu.Divider/>` - beta - allow overwriting internal dataHook

### PR DESCRIPTION
**Pull request by mykas**
This PR will expose dataHook prop for PopoverMenu beta compound component Menuitem. Fixing: https://github.com/wix/wix-style-react/issues/3821

### CODE REVIEW

File: src/beta/PopoverMenu/test/PopoverMenu.private.uni.driver.js [hunk 1]
```diff
@@ -3,12 +3,19 @@ import { dropdownBasePrivateDriverFactory } from '../../../DropdownBase/Dropdown
...
+        : `[data-hook="popover-menu-${option}"]`,
+    );
 
   return {
     ...publicDriverFactory(base),
     keyDownTrigger: key => keyDown(key),
     keyDownOption: async (option, key) => getOption(option).pressKey(key),
-    getMenuItem: async option => await getOption(option).getNative(),
+    getDivider: async dataHook => getOption('', dataHook).getNative(),
+    getMenuItem: async (option, dataHook) =>
```
Comment (splintor) [inline src/beta/PopoverMenu/test/PopoverMenu.private.uni.driver.js]: Why do we need to pass `option` as menuItem? `dataHook` should be enough.
You should check if `option` is a number and then decide if you want to use it as index (and search for `popover-menu-${option}`) or as `dataHook`.
Another option is to make it explicit and have two separate methods: `getMenuItemByIndex` and `getMenuItemByLabel`.

But having an additional `option` parameter that is ignored if `dataHook` is passed looks like a strange API.

Also, you need to update `PopoverMenu.uni.driver.js` and `PopoverMenu.uni.driver.d.ts` to make it possible to call `clickAtChild()` with `dataHook` and not just with `index`.

Comment (mykas) [reply src/beta/PopoverMenu/test/PopoverMenu.private.uni.driver.js]: Actually this is `private.uni.driver` which is not exposed. The method `clickAtChild()` is using DropdownBase infrastructure to click on the option by specifying the index number.

Comment (splintor) [reply src/beta/PopoverMenu/test/PopoverMenu.private.uni.driver.js]: Any plans to add `getChildByDataHook()` or `clickAtChildByDataHook()`?

Comment (mykas) [reply src/beta/PopoverMenu/test/PopoverMenu.private.uni.driver.js]: No probs, you can open a PR for this. I suggest reusing the `clickAtchild()` method to accept dataHook as second parameter. The first parameter should stay as `option`. There are users that uses in this way.

**Pull request merged**